Orthtree/doc/Orthtree/Concepts/OrthtreeTraitsWithData.h

```diff
@@ -0,0 +1,76 @@
+/*!
+  \ingroup PkgOrthtreeConcepts
+  \cgalConcept
+
+  The concept `OrthtreeTraitsWithData` defines the requirements for the
+  template parameter of the `CGAL::Orthtree` class for a node type that stores data.
+
+  \cgalRefines{OrthtreeTraits}
+
+  \cgalHasModelsBegin
+  \cgalHasModels{CGAL::Orthtree_traits_point<GeomTraits, PointRange, PointMap, dimension>}
+  \cgalHasModels{CGAL::Orthtree_traits_face_graph<PolygonMesh, VPM>}
+  \cgalHasModels{CGAL::Orthtree_traits_base<K, dimension>}
+  \cgalHasModelsEnd
+*/
+class OrthtreeTraitsWithData
+{
+public:
+
+  /// \name Types
+  /// @{
+  /*!
+   * \brief The data type contained by each node. Must be default constructible, copy constructible and copy assignable.
+   */
+  using Node_data = unspecified_type;
+
+  /*!
+   * \brief Functor which initializes elements contained by the root node.
+   *
+   * Each node of a tree has an associated `Node_data` value.
+   * This functor initializes the `Node_data` of the root node.
+   * It takes no arguments, and returns an instance of `Node_data`.
+   *
+   * Provides the operator:
+   * `Node_data operator()()`
+   *
+   * Typically, the `Node_data` of the root node contains all the elements in the tree.
+   * For a tree in which each node contains a span (such as `std::span()`) this function would return the span containing all items.
+   *
+   */
+  using Construct_root_node_contents = unspecified_type;
+
+  /*!
+   * \brief Functor which fills the contents of the nodes children.
+   *
+   * Provides the operator:
+   * `void operator()(typename Tree::Node_index, Tree&, const Point_d&)`
```

Comment:
How can the type `Tree` be known here?

Reply:
The type of the tree is Orthtree<Self> for the traits. The traits include the Orthtree.h

Reply:
Tree::Node_index is just an alias for OrthtreeTraits::Node_index

I am not sure how to write the type of the tree correctly. How to reference the traits themselves?
void operator()(Node_index, Orthtree<Self>&, const Point_d&)

Reply:
So it is then just `Node_index`.    But what about `Tree`?

Reply:
I think now I understand the problem. We do not speak about a member function of the traits class, but about the operator of a functor.

Reply:
Sorry, yes. A functor the traits have to provide that is called by the orthtree. As this functor needs to access nodes and their associated data, it needs an instance of the orthtree to operate on.

Reply:
It refines `OrthtreeTraits` so `Node_index` is defined. For the concept description you can use `Orthtree<Traits,b>`